File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.empty()) {
            return 0;
        }
        
        sort(nums.begin(),nums.end());
        int currentSequence = 1;
        int longestSequence = 0;

        for(int i = 1; i < nums.size(); i++) {
            if(nums[i] != nums[i - 1]) {
                if(nums[i] == nums[i - 1] + 1) {
                    currentSequence++;
                }
                else {
                    longestSequence = max(longestSequence, currentSequence);
                    currentSequence = 1;
                }
            }
        }
        return max(longestSequence, currentSequence);;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        numSet = set(nums)
        longest = 0

        for num in numSet:

            if (num - 1) not in numSet:
                currentNum = num
                currentStreak = 1

                while (currentNum + 1) in numSet:
                    currentNum += 1
                    currentStreak += 1
                
                longest = max(longest, currentStreak)

        return longest

            